SEED-platform/seed
# Feat/Email post office

#### Any background context you want to provide?
This is a feature by ClearlyEnergy, that we're moving to SEED from our tool BEAM.

#### What's this PR do?
Adds a post office to SEED. Users can create email templates in the organization settings, and then send emails en-masse to building owners. The email addresses are pulled from their owner_email field.

Email templates can include inventory fields using {{ field_name }}. For example, in a template you can write "Dear {{ first_name }}," and when sent en-masse the emails will be autofilled in from each building's information: "Dear John," "Dear Jane," et cetera.

The post office is fairly bare-bones, and there are more features I'd like to add in the future, such as a more detailed modal window in the inventory list, and a "preview email" button in the template page.

#### How should this be manually tested?
Templates can be created, edited, and deleted in the "Emails" organization tab. On the inventory list, we added an item for Emails to the "Actions" dropdown. You can select buildings in the list, navigate to Emails, and then select the template to send in the modal window. Emails will be created and added to a queue.

The post office may have to be configured further, to (for example) send emails on a schedule. See the [django-post_office page](https://github.com/ui/django-post_office) for more information.

The commits in this PR were cherry-picked from a long string of commits, so there _is_ a chance there are some code edits missing.

I wrote some quick Python tests in test_postoffice and test_views, but more unit testing should be done.

#### What are the relevant tickets?

#### Screenshots (if appropriate)
Templates page:
![image](https://user-images.githubusercontent.com/53191386/120849413-1377d300-c544-11eb-9048-a9803102ba9d.png)
